src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs

```diff
@@ -0,0 +1,203 @@
+                    {
+                        SqlParameter param = cmd.Parameters.Add("@c1", SqlDbType.Int);
+                        param.Value = DBNull.Value;
+                        cmd.ExecuteNonQuery();
+
+                        param.Value = 10;
+                        cmd.ExecuteNonQuery();
+                    }
+
+                    string sql1 = $"CREATE FUNCTION {UdfName}() RETURNS INT AS \n BEGIN \n RETURN (SELECT c1 FROM [{tableName}] WHERE c1 IS NULL)\n END";
+                    string sql2 = $"CREATE FUNCTION {UdfNameNotNull}() RETURNS INT AS \n BEGIN \n RETURN (SELECT c1 FROM [{tableName}] WHERE c1 IS NOT NULL)\n END";
+                    using (SqlCommand cmd = sqlConnection.CreateCommand())
+                    {
+                        cmd.CommandText = sql1;
+                        cmd.ExecuteNonQuery();
+
+                        cmd.CommandText = sql2;
+                        cmd.ExecuteNonQuery();
+                    }
+                }
+            }
+        }
+
+        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringSetupForAE))]
+        [ClassData(typeof(NullValueTestsData))]
+        public void NullValueTests(string connString, ConnStringColumnEncryptionSetting connStringSetting, SqlCommandColumnEncryptionSetting commandSetting, ReturnValueSetting nullReturnValue)
+        {
+            switch (connStringSetting)
+            {
+                case ConnStringColumnEncryptionSetting.Enabled:
+                    connString += "; Column Encryption Setting=Enabled";
+                    break;
+                case ConnStringColumnEncryptionSetting.Disabled:
+                    connString += "; Column Encryption Setting=Disabled";
+                    break;
+            }
+
+            using (SqlConnection sqlConn = new SqlConnection(connString))
+            {
+                sqlConn.Open();
+
+                // Create a command similarly
+                using (SqlCommand cmd = new SqlCommand(String.Format("SELECT c1 FROM [{0}] ORDER BY c2 ASC", tableName),
+                    sqlConn, null, commandSetting))
+                using (SqlCommand cmd2 = new SqlCommand((ReturnValueSetting.Null == nullReturnValue) ? UdfName : UdfNameNotNull,
+                    sqlConn, null, commandSetting))
+                {
+                    using (SqlDataReader reader = cmd.ExecuteReader())
+                    {
+                        Assert.True(reader.Read(), "No rows fetched first time");
+
+                        // Depending on the various flags verify the results
+                        object value1 = reader.GetProviderSpecificValue(0);
+
+                        // Read next row
+                        Assert.True(reader.Read(), "No rows fetched second time");
+                        object value2 = reader.GetProviderSpecificValue(0);
+
```

Comment:
You can now close `}` the reader's using. You don't need lines 96-98 anymore. You could even close `}` the cmd's using here, too. cmd2's using can be moved down to before line 100, too, and then closed before the switch statement.

Reply:
It sounds reasonable. I will change it and also move some of the variables' declaration out of `using` block.